### PR TITLE
Port visibility hidden for get_type_info methods

### PR DESCRIPTION
**Pull request by ilyachur**
### Details:
 - Port PR https://github.com/openvinotoolkit/openvino/pull/10691  to 2022.1.1 branch

### Tickets:
 - 82797

### CODE REVIEW

File: samples/cpp/hello_reshape_ssd/main.cpp [hunk 1]
```diff
@@ -53,7 +53,7 @@ int main(int argc, char* argv[]) {
         // try to find it.
         ov::NodeVector ops = model->get_ops();
         auto it = std::find_if(ops.begin(), ops.end(), [](std::shared_ptr<ov::Node> node) {
-            return node->get_type_info().name == ngraph::op::DetectionOutput::get_type_info_static().name;
+            return node->get_type_info() == ngraph::op::DetectionOutput::get_type_info_static();
```
Comment (ilya-lavrenov) [inline samples/cpp/hello_reshape_ssd/main.cpp:56]: why this change is needed? Is it not BW compatible change?

**Review by ilya-lavrenov — APPROVED** [review]

File: samples/cpp/hello_reshape_ssd/main.cpp (hunk 1 above)
Comment (ilyachur) [reply samples/cpp/hello_reshape_ssd/main.cpp:56]: The original code was incorrect. We cannot guarantee that we use the same pointers.

Comment (ilya-lavrenov) [reply samples/cpp/hello_reshape_ssd/main.cpp:56]: Where do we have pointers?
